### PR TITLE
refactor: extract registry logic

### PR DESCRIPTION
**Pull request by halber**
There is a registry logic that is commonly used. In order to reuse this logic, I have moved this logic to a new WriteSafeRegistry class.

### CODE REVIEW

File: src/main/java/io/neonbee/entity/EntityVerticle.java [hunk 1]
```diff
@@ -250,34 +255,19 @@ private Future<Void> announceEntityVerticle(Vertx vertx) {
...
-                                            });
-                                        });
+                                String qualifiedName = getQualifiedName();
+                                return getRegistry(vertx).register(name, qualifiedName);
                             }).collect(Collectors.toList());
                     return allComposite(announceFutures).mapEmpty();
                 });
     }
+
+    static WriteSafeRegistry getRegistry(Vertx vertx) {
```
Comment (s4heid) [inline src/main/java/io/neonbee/entity/EntityVerticle.java:267]: You ensure thread safety via the locking in the `WriteSafeRegistry` itself, but honestly I don't feel comfortable with this singleton approach. It comes with several other drawbacks like the fact that we need to introduce a `EntityVerticleExtension`. I would even rather manage it centrally via the NeonBee instance instead. Did you consider this?

File: src/main/java/io/neonbee/internal/WriteSafeRegistry.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    /**
+     * Unregister the value in the {@link NeonBee} async shared map from the sharedMapKey.
+     *
+     * @param sharedMapKey the shared map key
+     * @param value        the value to register
+     * @return the future
+     */
+    public Future<Void> unregister(String sharedMapKey, String value) {
```
Comment (s4heid) [inline src/main/java/io/neonbee/internal/WriteSafeRegistry.java:116]: this method is completely untested. Also noticed there is no `WriteSafeRegistryTest` class; might be a good idea to introduce and write a test for `unregister`.

File: src/main/java/io/neonbee/entity/EntityVerticle.java (hunk 1 above)
Comment (s4heid) [reply src/main/java/io/neonbee/entity/EntityVerticle.java:267]: Maybe, let's have a short sync after standup to discuss available options.

**Review by s4heid — CHANGES_REQUESTED** [review]

**Comment by pk-work** [comment]
Also from my side a big thank you @halber !

**Pull request merged**